docs/technology_stack.adoc

```diff
@@ -7,35 +7,43 @@
 
-    .my-class {
-      @include my-class;
-    }
+Task Runner::
+We use Gulp for running predefined tasks. These tasks include, compiling/preprocessing the LESS/TypeScript sources into CSS/JS, building the distributable artifacts, running dev-server in watch mode and general house-keeping (install, reinstall, cleanup etc.).
 
-We use mixins to avoid polluting components with unnecessary style classes, and to avoid
-an explosion of shared files.
+Build System::
+We use both Gulp and webpack for building the source. While gulp takes care of the library builds, webpack is used for building the (minimal) `*_./runtime_*` to use the library, which makes Planner work as a standalone app. The build system works as intended for now, but can be expanded to use more sophisticated switches to build according to different needs (dev, tests, prod etc.).
 
-The *_src/assets/stylesheets/_* directory includes a *_shared_* directory. These are
-shared global styles which will be refactored into a shared library at some point.
-You can update these styles only if you are making a truly global style, and are going to
-synchronize your changes across all the various UI projects.
+Test Framework::
+Our test stack is built with Karma, Protractor and PhantomJS. PhantomJS gives the headless browser engine to run the app, which Protractor interacts with as a WebDriverJS wrapper and runs the tests according to Karma's configurations.
 
-== Gulp
-<TBD>
+Each `*_filename.ts_*` file is supposed to have unit tests along with them as a `*_filename.spec.ts_*` in the same parent directory. The functional tests reside inside `*_./runtime/tests/_*`. For more details about running the tests, please look into link:./testing.adoc[Testing Documentation].
+
+CI/CD::
+The CI/CD pipeline is made with Jenkins, as defined in the `*_./Jenkinsfile_*` at the root of the repo. It uses the content of `*_./deploy_*` (more specifically, `*_./deploy/release.groovy_*`) to orcastrate the CI/CD.
+
+Release Framework::
+We use semantic-release for versioning and publishing our Planner library to npm registry. It's fairly configuration less CLI task (`$ npm run semantic-release`) which takes the latest PR/merges and commit messages to account. It generates the new version number, publishes to npm registry, and then generates the release tag on GitHub in one simple process.
+
+Deployment::
+At the end of the day Planner is a static library, and can be used/served with minimal static frontend runtime as a standalone app or with a full fledged Fabric8 UI. So whatever can make those setups run, can essentially be the deployment platform for Planner. We use Docker Compose to generate the image, Docker for running the built app container, and nginx as the web server to serve the app contents.
```

Comment:
Included this part to clarify that planner doesn't have any added burden over the runtime's deployment setup that it's integrated into. Can be removed, but I think that's an important context to set for someone going through the doc, otherwise it seems like the mentioned requirements are extra parts and pieces (and not something that's already needed for the runtime/base-app).